html_css/v2/foundations/css-foundations.md

```diff
@@ -0,0 +1,425 @@
+
+There's really no limit to how many combinators you can add to a rule, so `.one .two .three .four` would be totally valid. This would just select an element that has a class of `four` if it has an ancestor with a class of `three`, and if that ancestor has its own ancestor with a class of `two`, and so on. You generally want to avoid trying to select elements that require this level of nesting, though, as it can get pretty confusing and long, and it can cause issues when it comes to specificity. Speaking of which...
+
+### The Cascade of CSS
+
+Sometimes we may have rules that conflict with one another, and we end up with some unexpected results. "But I wanted *these* paragraphs to be blue, why are they red like these other paragraphs?!" As frustrating as this can be, it's important to understand that CSS doesn't just *do* things against our wishes. CSS only does what we tell it to do. One exception to this is the default styles that are provided by a browser. These default styles vary from browser to browser, and they are why some elements create a large "gap" between itself and other elements, or why buttons look the way they do, despite us not writing any CSS rules to style them that way.
+
+So if you end up with some unexpected behavior like this it's either because of these default styles, or due to not understanding how a property works or not understanding this little thing called the cascade. 
+
+The cascade is what determines which rules actually get applied to our HTML. There are different factors that the cascade uses to determine this, three of which we'll go over to hopefully help you avoid (as many of) those frustrating "I hate CSS" moments.
+
+#### Specificity
+
+Remember earlier how we mentioned that inline CSS overrides internal and external CSS? That's because inline CSS has a higher *specificity*. The style is directly on the element itself, and it's hard to get more specific than that.
+
+When it comes to internal and external CSS, each type of selector has its own specificity level (there are other selectors that make up the following list, but we're focusing on the ones mentioned in this lesson):
+
+1. ID selectors (most specific, after inline CSS)
+2. Class selectors
+3. Type selectors
+4. Universal selector and combinators (no specificity)
+
+Specificity will only be taken into account when an element has multiple rules targeting it, sort of like a tie-breaker. Ignoring universal selectors and combinators, the other types of selectors will increase the specificity of a rule for that selector's level.
+
+Specificity is represented by a comma-separated list of numbers: `0, 0, 0`. Starting from the left, each number represents the three specificity levels (id, class, and type), with each number having a higher precedence than the next (ID beats all, class beats type and no specificity, type beats anything left over). 
```

Comment:
I'm trying to decide if the scorecard is too far here.  On one hand, it might be more appropriate to give a couple of easy-to-remember rules like "ids beat classes beat types" and "more classes is more specific"

BUT: knowing exactly how it works is also valuable and I'm torn lol.

I think when it comes down to it... I personally never refer to or try to calculate the actual specificity score when i'm doing dev so I'd probably leave it out here.

I think if i had written this I might go with the simpler rules, and then say something like:
> if you want to know _exactly_ how specificity is calculated check this link here.

Reply:
The main reason I included the whole scorecard thing was an attempt to maybe simplify specificty for people at first. "100 is higher than 10, so ID wins", then once they get further along they won't need to use that anymore. Though at the same time yeah I do get that introducing something like this that people will most likely not need after long may not make much sense.

"X beats Y and Y, even if there are 100 Y's and only 1 X" or something would get the same point across.